pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -493,6 +498,10 @@ func fetchAvailabilityZones(client awsclient.Client, region string) ([]string, e
 func decodeAWSMachineProviderSpec(rawExt *runtime.RawExtension, scheme *runtime.Scheme) (*awsprovider.AWSMachineProviderConfig, error) {
 	codecFactory := serializer.NewCodecFactory(scheme)
 	decoder := codecFactory.UniversalDecoder(awsprovider.SchemeGroupVersion)
+	if rawExt == nil {
```

Comment:
This is redundant with the check added in `syncMachineSets`. My preference is to keep the check here and remove the check in `syncMachineSets`.